src/math/power.c

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+ /* Include Files */
+#include <sof/math/power.h>
+#include <sof/debug/panic.h>
+#include <sof/audio/format.h>
+/* define constant */
+#define POWER_MAX_LIMIT 0x8000
+
+/* Function Definitions */
+
+/**
+ * Arguments	: int32_t b
+ * base input range [-32 to +32]
+ *
+ *		: int32_t e
+ * exponent input range [ -3 to +3 ]
+ *
+ * Return Type	: int32_t
+ * power output range [-32768 to + 32768]
+ *
+ * +------------------+------------------+------------------+--------+--------+--------+
+ * | base	      | exponent	 | power(returntype)|	b    |	e     |	 p     |
+ * +----+-----+-------+----+-----+-------+----+----+--------+--------+--------+--------+
+ * |WLen| FLen|Signbit|WLen| FLen|Signbit|WLen|FLen|Signbit | Qformat| Qformat| Qformat|
+ * +----+-----+-------+----+-----+-------+----+----+--------+--------+--------+--------+
+ * | 32 | 25  |	 1    | 32 | 29	 |  1	 | 32 | 15 |   1    | 6.26   | 2.30   | 16.16  |
+ * +------------------+------------------+------------------+--------+--------+--------+
+ */
+int32_t power_int32(int32_t b, int32_t e)
+{
+	int32_t i;
+	int32_t k;
+	int32_t multiplier;
+	int32_t p;
+
+	p = POWER_MAX_LIMIT;
+
+	if (e < 0) {
+		e = -e;
+		assert(b); /* Panic if attempt divide by zero */
```

Comment:
can we make this a non fatal check at function entry with a trace error.

Reply:
OK